package.py

```diff
@@ -1047,6 +1047,7 @@ def prepare_command(args):
     hash_extra_paths = [p.format(path=tf_paths) for p in hash_extra_paths]
 
     content_hash = bpm.hash(hash_extra_paths)
+    content_hash.update(json.dumps(build_plan, sort_keys=True).encode())
```

Comment:
Thinking aloud... :)

`build_plan` contains _path_ which may lead to different `content_hash` depending on where this code is running. Is this ok?

Will it trigger new redeployment if running from different machines?

Reply:
Good questions.
I need to check all of them first.
Will tell when I'll be ready later.

Reply:
You are right by all questions, the whole build_plan can't be taken for hash. But there is a situation when valuable modifications in the build plan doesn't lead to a layer rebuilding and redeployment when a zip file already exists.
So there is the need to change strategy and take into account all important parts of build plan except of paths because
paths already got into account indirectly by files content and that seems should be enough. Such change requires significant code refactoring in the package.py

Reply:
I think we need to close this PR as unacceptable and open an issue for the problem.

Reply:
What if we let the user specify/update `hash_extra` value for that? For example, `hash_extra` can take into account the content of the file using Terraform's function like [filemd5](https://www.terraform.io/docs/configuration/functions/filemd5.html)?

Reply:
We introduced `hash_extra` as a temporary measure due to limitations of current package.py implementation and I thought that we will remove it in future when the implementation will be improved.
If there is a way how the user can refer to a file content from itself in the hash_extra this may work but this will look as a very strange hack.

Reply:
I've opened an issue so we can decline this PR.